cmd/snap/main_test.go

```diff
@@ -41,11 +44,59 @@ import (
 // Hook up check.v1 into the "go test" runner
 func Test(t *testing.T) { TestingT(t) }
 
+func openPty() (pty, tty *os.File, err error) {
+	p, err := os.OpenFile("/dev/ptmx", os.O_RDWR, 0)
```

Comment:
Too much fiddling below just for that one line. We have spread at hand for that sort of integration test. There are existing password tests there.